search/elastic.py

```diff
@@ -341,46 +342,68 @@ def field_property(field_name, field_value):
             )
             self._clear_mapping(doc_type)
 
-    def index(self, doc_type, body, **kwargs):
+    def index(self, doc_type, sources, **kwargs):
         """
-        Implements call to add document to the ES index
+        Implements call to add documents to the ES index
         Note the call to _check_mappings which will setup fields with the desired mappings
         """
-        id_ = body['id'] if 'id' in body else None
-
-        log.debug("indexing %s object with id %s", doc_type, id_)
-
-        self._check_mappings(doc_type, body)
 
         try:
-            self._es.index(
-                index=self.index_name,
-                doc_type=doc_type,
-                body=body,
-                id=id_,
-                ** kwargs
+            actions = []
+            for source in sources:
+                self._check_mappings(doc_type, source)
+                id_ = source['id'] if 'id' in source else None
+                log.debug("indexing %s object with id %s", doc_type, id_)
+                action = {
+                    "_index": self.index_name,
+                    "_type": doc_type,
+                    "_id": id_,
+                    "_source": source
+                }
+                actions.append(action)
+            response = bulk(
+                self._es,
+                actions,
+                stats_only=True,
+                **kwargs
             )
+            # bulk() returns a tuple with summary information
+            # number of successfully executed actions and number of errors if stats_only is set to True.
+            if response[1]:
+                raise exceptions.ElasticsearchException
```

Comment:
1. Shouldn't there be some message with the ElasticsearchException (how would you know what specifically went wrong)?
2. Nitpick: When you have situations where you're unpacking a tuple, it's clearer to do something like:

```python
_num_actions, num_errors = bulk(self._es, actions, stats_only=True, **kwargs)
```

As long as you put an underscore prefix to unused vars, pylint shouldn't complain about it.